In [107]:
# Final Project
"""
# Inspect: 
.head()
.tail()
.describe()
.info()

Cleaning: (finding from inspection)
removing NA
Drop columns
convert % value


Analyzing: Answering Hypothesis:

->Are the best rated chocolates are mostly from a specific region (bean origin)? #bar
->Is there an association between ratings and cocoa %? #line
->What percentage of Cocoa is most enjoyed # same as Q2, but plot pie chart!!
->Does the date of review effect the rating of the chocolate bar? #line
->Do the companies have a favorite ingredient combination, or are their recipes random? # bar
->What company consistently produces the best chocolate # bar
->Does the orgin of the cacao bean have a correlation to the country its sold in? # correlation heatmap


# graphs
# summary 


(Bonus)Modeling:Machine learning 


-----------------------------------------------------------------------------------------------------------
********************************THURSDAY*******************************
# First: summarize your findings in more lines (aim for 2 lines at least if possible)
# Second: Add title, y-axis, and x-axis names to chart.
# Create Streamlit app fully functional before break if possible

# After break, deploy our app by 2pm and make presentation for friday(2pm-2:40pm)
#If done early, I will show you some machine learning!!

# OPTIONAL (at the end): Try to tidy up the code by removing unwanted cell, and adding comments!!




"""

'\n# Inspect: \n.head()\n.tail()\n.describe()\n.info()\n\nCleaning: (finding from inspection)\nremoving NA\nDrop columns\nconvert % value\n\n\nAnalyzing: Answering Hypothesis:\n\n->Are the best rated chocolates are mostly from a specific region (bean origin)? #bar\n->Is there an association between ratings and cocoa %? #line\n->What percentage of Cocoa is most enjoyed # same as Q2, but plot pie chart!!\n->Does the date of review effect the rating of the chocolate bar? #line\n->Do the companies have a favorite ingredient combination, or are their recipes random? # bar\n->What company consistently produces the best chocolate # bar\n->Does the orgin of the cacao bean have a correlation to the country its sold in? # correlation heatmap\n\n\n# graphs\n(Bonus)Modeling:Machine learning \n\n\n-----------------------------------------------------------------------------------------------------------\n********************************THURSDAY*******************************\n# First: summarize you

In [108]:
import pandas as pd

In [109]:
import plotly.express as px

In [110]:
import plotly.figure_factory as ff

<span style='font-size:xx-large'>**Inspection:** </span>


In [111]:
chocolate_df = pd.read_csv("Chocolate bar ratings 2022.csv")

In [112]:
chocolate_df.head()

REF Company (Manufacturer) Company Location  Review Date  \
0  2454                   5150           U.S.A.         2019   
1  2454                   5150           U.S.A.         2019   
2  2458                   5150           U.S.A.         2019   
3  2542                   5150           U.S.A.         2021   
4  2542                   5150           U.S.A.         2021   

  Country of Bean Origin Specific Bean Origin or Bar Name Cocoa Percent  \
0               Tanzania            Kokoa Kamili, batch 1           76%   
1             Madagascar           Bejofo Estate, batch 1           76%   
2     Dominican Republic                  Zorzal, batch 1           76%   
3                   Fiji            Matasawalevu, batch 1           68%   
4                  India                Anamalai, batch 1           68%   

  Ingredients Most Memorable Characteristics  Rating  
0    3- B,S,C      rich cocoa, fatty, bready    3.25  
1    3- B,S,C   cocoa, blackberry, full body    3.75  
2    3- B,S,C         cocoa, vegetal, savory    3.50  
3    3- B,S,C            chewy, off, rubbery    3.00  
4    3- B,S,C  milk brownie, macadamia,chewy    3.50

In [113]:
chocolate_df.tail()

REF Company (Manufacturer) Company Location  Review Date  \
2583  1205                 Zotter          Austria         2014   
2584  1996                 Zotter          Austria         2017   
2585  2036                 Zotter          Austria         2018   
2586  2170                 Zotter          Austria         2018   
2587  2170                 Zotter          Austria         2018   

     Country of Bean Origin Specific Bean Origin or Bar Name Cocoa Percent  \
2583                  Blend                              Raw           80%   
2584               Colombia                 APROCAFA, Acandi           75%   
2585                  Blend  Dry Aged, 30 yr Anniversary bar           75%   
2586                  Congo            Mountains of the Moon           70%   
2587                 Belize                         Maya Mtn           72%   

       Ingredients Most Memorable Characteristics  Rating  
2583  4- B,S*,C,Sa         waxy, cloying, vegetal    2.75  
2584      3- B,S,C      strong nutty, marshmallow    3.75  
2585      3- B,S,C           fatty, earthy, cocoa    3.00  
2586      3- B,S,C   fatty, mild nuts, mild fruit    3.25  
2587      3- B,S,C      muted, roasty, accessible    3.50

In [114]:
chocolate_df.describe()

REF  Review Date       Rating
count  2588.000000  2588.000000  2588.000000
mean   1459.185471  2014.545209     3.198319
std     773.852201     4.082722     0.444069
min       5.000000  2006.000000     1.000000
25%     820.000000  2012.000000     3.000000
50%    1470.000000  2015.000000     3.250000
75%    2122.000000  2018.000000     3.500000
max    2768.000000  2022.000000     4.000000

In [115]:
chocolate_df.columns

Index(['REF', 'Company (Manufacturer)', 'Company Location', 'Review Date',
       'Country of Bean Origin', 'Specific Bean Origin or Bar Name',
       'Cocoa Percent', 'Ingredients', 'Most Memorable Characteristics',
       'Rating'],
      dtype='object')

In [116]:
chocolate_df.isna().sum()

REF                                  0
Company (Manufacturer)               0
Company Location                     0
Review Date                          0
Country of Bean Origin               0
Specific Bean Origin or Bar Name     0
Cocoa Percent                        0
Ingredients                         87
Most Memorable Characteristics       0
Rating                               0
dtype: int64

In [117]:
chocolate_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2588 entries, 0 to 2587
Data columns (total 10 columns):
 #   Column                            Non-Null Count  Dtype  
---  ------                            --------------  -----  
 0   REF                               2588 non-null   int64  
 1   Company (Manufacturer)            2588 non-null   object 
 2   Company Location                  2588 non-null   object 
 3   Review Date                       2588 non-null   int64  
 4   Country of Bean Origin            2588 non-null   object 
 5   Specific Bean Origin or Bar Name  2588 non-null   object 
 6   Cocoa Percent                     2588 non-null   object 
 7   Ingredients                       2501 non-null   object 
 8   Most Memorable Characteristics    2588 non-null   object 
 9   Rating                            2588 non-null   float64
dtypes: float64(1), int64(2), object(7)
memory usage: 202.3+ KB


In [118]:
chocolate_df.shape

(2588, 10)

# cleaning:

drops

change type of cocoa percent to float


remove rows with null ingredient 


In [119]:
column_to_drop = ['REF']
chocolate_df.drop(column_to_drop, axis = 1, inplace = True)

In [120]:
chocolate_df.dropna(inplace=True)
chocolate_df.reset_index(drop=True, inplace=True)
chocolate_df.shape

(2501, 9)

In [121]:
chocolate_df['Cocoa Percent']=chocolate_df['Cocoa Percent'].str.replace('%','').astype(float)

In [122]:
chocolate_df.head()

Company (Manufacturer) Company Location  Review Date Country of Bean Origin  \
0                   5150           U.S.A.         2019               Tanzania   
1                   5150           U.S.A.         2019             Madagascar   
2                   5150           U.S.A.         2019     Dominican Republic   
3                   5150           U.S.A.         2021                   Fiji   
4                   5150           U.S.A.         2021                  India   

  Specific Bean Origin or Bar Name  Cocoa Percent Ingredients  \
0            Kokoa Kamili, batch 1           76.0    3- B,S,C   
1           Bejofo Estate, batch 1           76.0    3- B,S,C   
2                  Zorzal, batch 1           76.0    3- B,S,C   
3            Matasawalevu, batch 1           68.0    3- B,S,C   
4                Anamalai, batch 1           68.0    3- B,S,C   

  Most Memorable Characteristics  Rating  
0      rich cocoa, fatty, bready    3.25  
1   cocoa, blackberry, full body    3.75  
2         cocoa, vegetal, savory    3.50  
3            chewy, off, rubbery    3.00  
4  milk brownie, macadamia,chewy    3.50

In [123]:
chocolate_df.describe()

Review Date  Cocoa Percent       Rating
count  2501.000000    2501.000000  2501.000000
mean   2014.660136      71.472411     3.211815
std       4.071615       5.122815     0.427366
min    2006.000000      42.000000     1.000000
25%    2012.000000      70.000000     3.000000
50%    2015.000000      70.000000     3.250000
75%    2018.000000      74.000000     3.500000
max    2022.000000     100.000000     4.000000

In [124]:
chocolate_df.isna().sum()

Company (Manufacturer)              0
Company Location                    0
Review Date                         0
Country of Bean Origin              0
Specific Bean Origin or Bar Name    0
Cocoa Percent                       0
Ingredients                         0
Most Memorable Characteristics      0
Rating                              0
dtype: int64

<span style='font-size:xx-large'>**Team 1**</span>

\-&gt;Are the best rated chocolates are mostly from a specific region \(bean origin\)? <u>\#bar</u>

\-&gt;Is there an association between ratings and cocoa %? <u>\#line</u>

\-&gt;What percentage of Cocoa is most enjoyed \# same as Q2, but plot <u>pie chart!!</u>



In [125]:
#->Is there an association between ratings and cocoa %? (Line)

In [126]:
def company_fav_ingredient(df):
    company_ing = pd.DataFrame()
    for i in df['Ingredients'].unique().tolist():

        column = df[df["Ingredients"] == i]

        column = column.groupby('Company (Manufacturer)')['Company (Manufacturer)'].count().sort_values(ascending=False)
        column = column[:3]
        #company_ing[i] = column.values
        t = {}
        t[i] = column
        company_ing = pd.concat([company_ing, pd.DataFrame.from_records([column])])
    return company_ing

In [127]:
chocolate_df['Rating'].unique().shape


(12,)

In [128]:
def rating(data):
    tempdf = data
    finaldf = pd.DataFrame()
    for i in tempdf['Rating'].unique().tolist():
        column = tempdf[tempdf['Rating'] == i]
        cocoamean = column["Cocoa Percent"].mean()
        data1 = {
            "Cocoa %": [cocoamean],
            "Rating": [i]
                }
        if cocoamean >=0:
            finaldf = pd.concat([finaldf,pd.DataFrame.from_records(data1)],ignore_index=True)
    return finaldf

In [129]:
table = rating(chocolate_df)
table.head()

Cocoa %  Rating
0  71.406048    3.25
1  71.078073    3.75
2  71.214536    3.50
3  71.694391    3.00
4  71.840491    2.75

In [130]:
table = table.sort_values("Rating")
table

Cocoa %  Rating
11  73.000000    1.00
10  75.600000    1.50
7   72.000000    1.75
9   73.689655    2.00
8   72.928571    2.25
6   71.694969    2.50
4   71.840491    2.75
3   71.694391    3.00
0   71.406048    3.25
2   71.214536    3.50
1   71.078073    3.75
5   70.763158    4.00

In [131]:
import plotly.express as px
import numpy as np

In [132]:
fig = px.line(table, x= 'Rating',y='Cocoa %',title="Relationship Between Chocolate Ratings and Cocoa Percentage")
fig.show()

In [133]:
#As the rating of the chocolate goes up, on average, the percentage of cocoa used in the chocolate decreases.

In [134]:
#->What percentage of Cocoa is most enjoyed # (Pie chart)
#wedge sizes: max value of rating (array)
#labels: cocoa % (list)

In [135]:
chocolate_df['Rating'].unique()

array([3.25, 3.75, 3.5 , 3.  , 2.75, 4.  , 2.5 , 1.75, 2.25, 2.  , 1.5 ,
       1.  ])

In [136]:
toprating = chocolate_df[chocolate_df['Rating'] == 4.]
toprating

Company (Manufacturer) Company Location  Review Date  \
16                  A. Morin           France         2013   
19                  A. Morin           France         2013   
24                  A. Morin           France         2014   
32                  A. Morin           France         2021   
81                     Amano           U.S.A.         2010   
...                      ...              ...          ...   
2371                Valrhona           France         2012   
2374                Valrhona           France         2015   
2416  White Label aka Mutari           U.S.A.         2022   
2430          Willie's Cacao             U.K.         2016   
2437               Woodblock           U.S.A.         2011   

     Country of Bean Origin          Specific Bean Origin or Bar Name  \
16                Venezuela                                     Chuao   
19                     Peru                      Chanchamayo Province   
24                     Peru                                   Pablino   
32                   Mexico                                   La Joya   
81                  Ecuador                                    Guayas   
...                     ...                                       ...   
2371              Venezuela                         Otucan, Grand Cru   
2374                   Peru                  Piura, Illanka, Quemazon   
2416                Ecuador  Costa Esmeraldas, CEconche77121620121820   
2430               Colombia                                Los Llanos   
2437              Venezuela                                   Ocumare   

      Cocoa Percent   Ingredients   Most Memorable Characteristics  Rating  
16             70.0    4- B,S,C,L    oily, nut, caramel, raspberry     4.0  
19             63.0      3- B,S,C          sweet, cocoa, tangerine     4.0  
24             70.0    4- B,S,C,L      delicate, hazelnut, brownie     4.0  
32             70.0    4- B,S,C,L       light color, fruit, yogurt     4.0  
81             70.0    4- B,S,C,V     strong spice, intense pepper     4.0  
...             ...           ...                              ...     ...  
2371           69.0  5- B,S,C,V,L     green,unripened fruit,banana     4.0  
2374           63.0  5- B,S,C,V,L      intense, dark berry, grapes     4.0  
2416           77.0      3- B,S,C  complex, hazelnut, dairy, fruit     4.0  
2430           70.0      3- B,S,C       rich cocoa, spicy, cinamon     4.0  
2437           70.0        2- B,S              nutty, tart, coffee     4.0  

[114 rows x 9 columns]

In [137]:
table.head()

Cocoa %  Rating
11  73.000000    1.00
10  75.600000    1.50
7   72.000000    1.75
9   73.689655    2.00
8   72.928571    2.25

In [138]:
title = dict(text="Best Rated Cocoa %", position='top center')
fig = px.pie(table, values= 'Rating',names = 'Cocoa %', title= 'Best Rated Cocoa %', )
fig.show()

In [139]:
"""

"""

'\n\n'

# Team 2



\-&gt;\*\*\*Does the date of review effect the rating of the chocolate bar? #line
\-&gt;\*\*\*Do the companies have a favorite ingredient combination, or are their recipes random? \# bar
\-&gt;\*\*\*What company consistently produces the best chocolate \# bar



In [140]:
ratingsByYear = []
years = range(2006,2023)
for i in years:
    tempdf = chocolate_df[chocolate_df['Review Date']==i]
    ratingsByYear.append(tempdf['Rating'].mean())
data= list(zip(years,ratingsByYear))
myDF = pd.DataFrame(data, columns=['Year of Review','Rating'])
myDF

Year of Review    Rating
0             2006  3.062500
1             2007  3.166667
2             2008  3.038690
3             2009  3.088496
4             2010  3.188830
5             2011  3.271242
6             2012  3.206944
7             2013  3.214689
8             2014  3.197531
9             2015  3.243750
10            2016  3.238263
11            2017  3.364078
12            2018  3.191111
13            2019  3.134715
14            2020  3.256173
15            2021  3.320000
16            2022  3.284483

In [163]:
px.line(myDF, x = 'Year of Review', y = 'Rating', template = 'plotly_dark', title = 'Average Rating by Year', labels = {'Rating': 'Average Rating', 'Year of Review': "Year of Review Date"})

The relationship between YoR \(Year of Review\) and Rating is loosely proportional. As the Rating increases from 3.06 in 2006 to 3.28 in 2022, the correlation is positive. The percentage change between the start and end dates \(2006\-2022\) is approximately 7%. Unfortunately, the minimal amount of change demonstrated paired with the sporadic data leans towards the conclusion that YoR has minimal effect on the rating distributed.



In [142]:
chocolate_df.groupby('Rating')['Rating'].count().sort_values(ascending = False)

Rating
3.50    571
3.00    517
3.25    463
2.75    326
3.75    301
2.50    159
4.00    114
2.00     29
2.25     14
1.50      5
1.00      1
1.75      1
Name: Rating, dtype: int64

In [143]:
chocolate_df['Company (Manufacturer)'].unique().shape

(555,)

In [144]:
chocolate_df['Rating'].unique().shape

(12,)

In [145]:
def rc(data):   
    ratingComp = pd.DataFrame()
    for i in data['Rating'].unique().tolist():
        column = data[data['Rating'] == i]
        column = column.groupby('Company (Manufacturer)')['Company (Manufacturer)'].count().sort_values(ascending = False)
        column = column[:2]
        t = {}
        t[i] = column
        ratingComp= pd.concat([ratingComp, pd.DataFrame.from_records([column])])
    return ratingComp


In [146]:
df_rc = rc(chocolate_df)
l = chocolate_df['Rating'].unique().tolist()
df_rc['Rating'] = l

fig = px.bar(df_rc, x = df_rc["Rating"], y = df_rc.columns.tolist(), template = 'plotly_dark', title = 'Ratings by Company', labels = {'Rating': 'Chocolate Bar Ratings', 'value' : 'frequency'})
fig.show()

Company Fav Ingredient

\*Do the companies have a favorite ingredient combination, or are their recipes random? \# bar



In [147]:
chocolate_df.head()

Company (Manufacturer) Company Location  Review Date Country of Bean Origin  \
0                   5150           U.S.A.         2019               Tanzania   
1                   5150           U.S.A.         2019             Madagascar   
2                   5150           U.S.A.         2019     Dominican Republic   
3                   5150           U.S.A.         2021                   Fiji   
4                   5150           U.S.A.         2021                  India   

  Specific Bean Origin or Bar Name  Cocoa Percent Ingredients  \
0            Kokoa Kamili, batch 1           76.0    3- B,S,C   
1           Bejofo Estate, batch 1           76.0    3- B,S,C   
2                  Zorzal, batch 1           76.0    3- B,S,C   
3            Matasawalevu, batch 1           68.0    3- B,S,C   
4                Anamalai, batch 1           68.0    3- B,S,C   

  Most Memorable Characteristics  Rating  
0      rich cocoa, fatty, bready    3.25  
1   cocoa, blackberry, full body    3.75  
2         cocoa, vegetal, savory    3.50  
3            chewy, off, rubbery    3.00  
4  milk brownie, macadamia,chewy    3.50

In [148]:
chocolate_df.describe()

Review Date  Cocoa Percent       Rating
count  2501.000000    2501.000000  2501.000000
mean   2014.660136      71.472411     3.211815
std       4.071615       5.122815     0.427366
min    2006.000000      42.000000     1.000000
25%    2012.000000      70.000000     3.000000
50%    2015.000000      70.000000     3.250000
75%    2018.000000      74.000000     3.500000
max    2022.000000     100.000000     4.000000

In [149]:
chocolate_df.columns

Index(['Company (Manufacturer)', 'Company Location', 'Review Date',
       'Country of Bean Origin', 'Specific Bean Origin or Bar Name',
       'Cocoa Percent', 'Ingredients', 'Most Memorable Characteristics',
       'Rating'],
      dtype='object')

In [150]:
chocolate_df['Company (Manufacturer)'].unique().shape

(555,)

In [151]:
chocolate_df['Ingredients'].unique().shape

(21,)

In [152]:
def company_fav_ingredient(df):
    company_ing = pd.DataFrame()
    for i in df['Ingredients'].unique().tolist():

        column = df[df["Ingredients"] == i]

        column = column.groupby('Company (Manufacturer)')['Company (Manufacturer)'].count().sort_values(ascending=False)
        column = column[:3]
        #company_ing[i] = column.values
        t = {}
        t[i] = column
        company_ing = pd.concat([company_ing, pd.DataFrame.from_records([column])])
    return company_ing

In [166]:
df_cfi = company_fav_ingredient(chocolate_df)
l = chocolate_df['Ingredients'].unique().tolist()
df_cfi["Ingredients"] = l

fig = px.bar(df_cfi, x = df_cfi["Ingredients"], y = df_cfi.columns.tolist(),title = "Company Favorite Ingredients ", labels = {'Ingredients': 'Chocolate Ingredients', 'value' : 'Company'})
fig.show()

In [154]:
column = chocolate_df[chocolate_df['Ingredients'] == '3- B,S,C']
column

Company (Manufacturer) Company Location  Review Date  \
0                      5150           U.S.A.         2019   
1                      5150           U.S.A.         2019   
2                      5150           U.S.A.         2019   
3                      5150           U.S.A.         2021   
4                      5150           U.S.A.         2021   
...                     ...              ...          ...   
2478      Zoto (Chocolatoa)          Belgium         2018   
2497                 Zotter          Austria         2017   
2498                 Zotter          Austria         2018   
2499                 Zotter          Austria         2018   
2500                 Zotter          Austria         2018   

     Country of Bean Origin Specific Bean Origin or Bar Name  Cocoa Percent  \
0                  Tanzania            Kokoa Kamili, batch 1           76.0   
1                Madagascar           Bejofo Estate, batch 1           76.0   
2        Dominican Republic                  Zorzal, batch 1           76.0   
3                      Fiji            Matasawalevu, batch 1           68.0   
4                     India                Anamalai, batch 1           68.0   
...                     ...                              ...            ...   
2478              Nicaragua                           Rugoso           75.0   
2497               Colombia                 APROCAFA, Acandi           75.0   
2498                  Blend  Dry Aged, 30 yr Anniversary bar           75.0   
2499                  Congo            Mountains of the Moon           70.0   
2500                 Belize                         Maya Mtn           72.0   

     Ingredients Most Memorable Characteristics  Rating  
0       3- B,S,C      rich cocoa, fatty, bready    3.25  
1       3- B,S,C   cocoa, blackberry, full body    3.75  
2       3- B,S,C         cocoa, vegetal, savory    3.50  
3       3- B,S,C            chewy, off, rubbery    3.00  
4       3- B,S,C  milk brownie, macadamia,chewy    3.50  
...          ...                            ...     ...  
2478    3- B,S,C             sandy, basic cocoa    3.25  
2497    3- B,S,C      strong nutty, marshmallow    3.75  
2498    3- B,S,C           fatty, earthy, cocoa    3.00  
2499    3- B,S,C   fatty, mild nuts, mild fruit    3.25  
2500    3- B,S,C      muted, roasty, accessible    3.50  

[1029 rows x 9 columns]

# Correlation Heatmap



In [155]:
chocolate_df.columns

Index(['Company (Manufacturer)', 'Company Location', 'Review Date',
       'Country of Bean Origin', 'Specific Bean Origin or Bar Name',
       'Cocoa Percent', 'Ingredients', 'Most Memorable Characteristics',
       'Rating'],
      dtype='object')

In [156]:
tempdf = chocolate_df.copy()
tempdf.drop(['Company (Manufacturer)', 'Company Location', 'Country of Bean Origin', 'Specific Bean Origin or Bar Name','Ingredients', 'Most Memorable Characteristics'], axis =1, inplace = True)

chocolate_df_corr = tempdf.corr() # Generate correlation matrix
x = list(chocolate_df_corr.columns)
y = list(chocolate_df_corr.index)
z = np.array(chocolate_df_corr)

fig = ff.create_annotated_heatmap(
    z,
    x = x,
    y = y ,
    annotation_text = np.around(z, decimals=2),
    hoverinfo='z',
    colorscale='Brwnyl',
    showscale=True,
    )
fig.update_xaxes(side="bottom")
fig.update_layout(
    # title_text='Heatmap', 
    title_x=0.5, 
    width=500, 
    height=500,
    yaxis_autorange='reversed',
    template='plotly_dark'
)
fig.show()

In [157]:
df_temp = chocolate_df.copy()
df_temp['Country of Bean Origin']=df_temp['Country of Bean Origin'].astype('category').cat.codes
df_temp['Company Location']=df_temp['Company Location'].astype('category').cat.codes

df_temp.drop(['Company (Manufacturer)', 'Specific Bean Origin or Bar Name','Ingredients', 'Most Memorable Characteristics'], axis =1, inplace = True)

chocolate_df_corr = df_temp.corr() # Generate correlation matrix
x = list(chocolate_df_corr.columns)
y = list(chocolate_df_corr.index)
z = np.array(chocolate_df_corr)

fig = ff.create_annotated_heatmap(
    z,
    x = x,
    y = y ,
    annotation_text = np.around(z, decimals=2),
    hoverinfo='z',
    colorscale='Brwnyl',
    showscale=True,
    )
fig.update_xaxes(side="bottom")
fig.update_layout(
    # title_text='Heatmap', 
    title_x=0.5, 
    width=500, 
    height=500,
    yaxis_autorange='reversed',
    template='plotly_dark'
)
fig.show()

In [158]:
chocolate_df.head()

Company (Manufacturer) Company Location  Review Date Country of Bean Origin  \
0                   5150           U.S.A.         2019               Tanzania   
1                   5150           U.S.A.         2019             Madagascar   
2                   5150           U.S.A.         2019     Dominican Republic   
3                   5150           U.S.A.         2021                   Fiji   
4                   5150           U.S.A.         2021                  India   

  Specific Bean Origin or Bar Name  Cocoa Percent Ingredients  \
0            Kokoa Kamili, batch 1           76.0    3- B,S,C   
1           Bejofo Estate, batch 1           76.0    3- B,S,C   
2                  Zorzal, batch 1           76.0    3- B,S,C   
3            Matasawalevu, batch 1           68.0    3- B,S,C   
4                Anamalai, batch 1           68.0    3- B,S,C   

  Most Memorable Characteristics  Rating  
0      rich cocoa, fatty, bready    3.25  
1   cocoa, blackberry, full body    3.75  
2         cocoa, vegetal, savory    3.50  
3            chewy, off, rubbery    3.00  
4  milk brownie, macadamia,chewy    3.50

<span style='font-size:xx-large'>Analysis</span>


# <span style='font-size:x-large'>Team 1: </span>



**Hypothesis 1:** What percentage of Cocoa is most enjoyed?

\-The ratings for different cocoa percentages varied from 1 to 4.
\-Out of these ratings, the best ratings were distributed mostly with chocolates with the lowest Cocoa Percentage
\-Between the top two Cocoa Percentages, there is a 0.8% difference in the distribution of best ratings, but there was a clear winner of 70.76%



**Hypothesis 2**: Is there an association between ratings and cocoa %?

 - On average, as the rating of the chocolate goes up, the percentage of cocoa used in the chocolate decreases.
 - The highest average % of cocoa which was at 75.6% sat at a 1.5 rating while the lowest average % of cocoa which was 70.76% reached a full a 4.0 rating.
     • In comparison, the lowest rated chocolates which were at a 1.0 rating, had a 73% cocoa makeup on average.
 - From a rating of 2.5 and onwards, the average % of cocoa used in chocolate maintained a value below 72%
 - The percentage of cocoa used in chocolate appears to have a negative effect on the rating given to the chocolate.



## Team 2



**Hypothesis 3:** Does the date of review effect the rating of the chocolate bar?

The relationship between YoR \(Year of Review\) and Rating is loosely proportional. As the Rating increases from 3.06 in 2006 to 3.28 in 2022, the correlation is positive. The percentage change between the start and end dates \(2006\-2022\) is approximately 7%. Unfortunately, the minimal amount of change demonstrated paired with the sporadic data leads towards the conclusion that YoR has minimal effect on the rating distributed.



**Hypothesis 4:** What company consistently produces the best chocolate?

We can consider the best chocolate to be ranked 3.25 or above. Based on the Bar Chart titled Ratings by Company, Soma is the only company that is consistent within the ratings of 3.25 \- 4. Making Soma the most consitent producer of the best chocolate. 

However Soma is no the only company that is consitent within the ratings of 3.25\-4. Fresco is also consistent, but only within ratings of 3.25\-3.5. Since that range is above 3.25, Fresco's chocolate qualifies as one of the best chocolate. This makes Fresco, a runner up to Soma as the most consitent producer of the best chocolate.



**Hypothesis 5:** Does the origin of the cacao bean have a correlation to the country its sold in?

